sympy/solvers/polysys.py

```diff
@@ -816,3 +818,258 @@ def _poly_sort_key(poly):
+
+
+def dom_get_alg_lift(K):
+    if K.is_ZZ_I or K.is_QQ_I:
+        dom = K.dom
+        lift = lambda e: _dom_lift_gaussian(e, K)
+        lower = lambda l: _dom_lower_gaussian(l, K)
+    elif K.is_AlgebraicField:
+        dom = K.dom
+        lift = lambda e: _dom_lift_algebraic(e, K)
+        lower = lambda l: _dom_lower_algebraic(l, K)
+    elif K.is_PolynomialRing:
+        dom_e, lift_e, lower_e = dom_get_alg_lift(K.dom)
+        dom = dom_e[K.symbols]
+        lift = lambda e: _dom_lift_poly(e, dom, lift_e)
+        lower = lambda l: _dom_lower_poly(l, K, lower_e)
+    else:
+        raise NotImplementedError
+    return dom, lift, lower
+
+
+def _lift_poly(poly: Poly, new_gen: Symbol) -> Poly:
+    domain = poly.domain
+    dom, lift, _ = dom_get_alg_lift(domain)
+
+    new_gens = poly.gens + (new_gen,)
+    new_coeffs = {}
+
+    for monom, coeff in poly.rep.to_dict().items():
+        lifted = lift(coeff)
+        for i, c in enumerate(lifted):
+            new_monom = monom + (i,)
+            new_coeffs[new_monom] = c
+
+    return Poly.from_dict(new_coeffs, *new_gens, domain=dom)
+
+
+def _lower_poly(poly: Poly, orig_domain, gen_index: int) -> Poly:
+    _, _, lower = dom_get_alg_lift(orig_domain)
+
+    coeffs = defaultdict(list)
+    for monom, coeff in poly.rep.to_dict().items():
+        base_monom = monom[:gen_index] + monom[gen_index + 1:]
+        deg = monom[gen_index]
+        while len(coeffs[base_monom]) <= deg:
+            coeffs[base_monom].append(0)
+        coeffs[base_monom][deg] = coeff
+
+    new_coeffs = {}
+    for monom, coeff_list in coeffs.items():
+        if not coeff_list:
+            coeff_list = [0, 0]
+        elif len(coeff_list) == 1:
+            coeff_list.append(0)
+        new_coeffs[monom] = lower(coeff_list)
+
+    return Poly.from_dict(new_coeffs, *poly.gens[:gen_index], domain=orig_domain)
+
+
+def groebner_basis_zz(polys: list[Poly], gens: Tuple) -> list[Poly]:
+    exprs = [p.as_expr() for p in polys]
+    basis = groebner(exprs, gens)
+    return [Poly(b, *gens, domain=ZZ) for b in basis]
+
+
+def groebner_basis_qq(polys: list[Poly], gens: Tuple) -> list[Poly]:
+    cleared = [p.clear_denoms()[1] for p in polys]
+    basis = groebner_basis_zz(cleared, gens)
+    return [p.set_domain(QQ) for p in basis]
+
+
+def groebner_basis_complex(polys: list[Poly], gens: Tuple) -> list[Poly]:
+    domain = polys[0].domain
+    i = Symbol('i')
+    new_gens = gens + (i,)
+
+    if domain.is_QQ_I:
+        polys = [p.clear_denoms()[1] for p in polys]
+
+    lifted_polys = [_lift_poly(p, i) for p in polys]
+    min_poly = Poly(i ** 2 + 1, *new_gens, domain=ZZ)
+    lifted_polys.append(min_poly)
+
+    basis = groebner([p.as_expr() for p in lifted_polys], new_gens)
+
+    result = []
+    for b in basis:
+        poly = Poly(b, *new_gens, domain=ZZ)
+        if i not in poly.free_symbols:
+            result.append(_lower_poly(poly, domain, len(gens)))
```

Comment:
If you don't use `as_expr` then `groebner` will return these as `Poly`.

It shouldn't be necessary to check `if i not in poly.free_symbols` because the equation involving `i` will always be first or last. A better check would be to check the degree of the polynomials in `i`. There should be exactly one polynomial of degree 2 in `i` and every other should be of degree 1 or less.